### ISSUE TITLE
Implicit return is undetected

### ISSUE THREAD

**Issue by james2doyle** [gajus/eslint-plugin-jsdoc#148]
![screen shot 2019-01-16 at 2 30 51 pm](https://user-images.githubusercontent.com/1425304/51282969-81cf9100-199b-11e9-9b2c-8f6158b11cbe.png)

As you can see, implicit returns are not properly detected. It is saying that there is a `@returns` but there shouldn't be. This is incorrect as the implicit return does not require a `return` keyword (seems like this is what is being detected) to work properly.

Any ideas on where this could be fixed? I can make a PR but some direction would be helpful

**Comment by ptkdev** [comment]
In #147 pull request i split rule, if you disable  `require-returns-check` this error not appear.
If you enable `require-returns` if exist return string, error appear. Wait merge and try if work.

**Comment by james2doyle** [comment]
@ptkdev that PR is merged. So I don’t think it has fixed it.

Looking at the source, just doing "index of return" is never going to work when there is an implicit return

**Comment by gajus** [comment]
This is current not being check. PR is welcome.

**Comment by ptkdev** [comment]
In **`v4.0.0`** only solution for this bug is disable `require-returns-check`.
I don't know how detect implicit with eslint api. Imho this is hard.

**Comment by james2doyle** [comment]
I will see if I can get it to work. Or at least improve the current `return` detect as it also detects deep returns that are not at the top level

**Comment by gajus** [comment]
> Or at least improve the current return detect as it also detects deep returns that are not at the top level

If thats really the case, then I would appreciate if you create an issue for it.

**Comment by james2doyle** [comment]
Yeah the code just looks for any "return" keyword in the code body. I will see if I can refactor that to be a little more accurate. 

There are features in eslint that handle implicit return, so I'm sure their code should help indicate a potential solution: https://eslint.org/docs/rules/arrow-body-style

**Comment by garyking** [comment]
For `jsdoc/require-returns`, please detect implicit returns, and also, don't apply the rule to constructors which have docs, since they can't return anything.

**Comment by james2doyle** [comment]
Opened https://github.com/gajus/eslint-plugin-jsdoc/pull/155 for people who are watching this and looking for a fix
